src/test/groovy/edu/ie3/datamodel/models/input/NodeInputTest.groovy

```diff
@@ -32,4 +35,25 @@ class NodeInputTest extends Specification {
 			assert voltLvl == GermanVoltageLevelUtils.EHV_220KV
 		}
 	}
+
+	def "A NodeInput equals method should work as expected"() {
+
+		when:
+		nodeA
+		nodeB
+
+		then:
+		nodeA.equals(nodeB) == expectedResult
+
+		where:
+		nodeA                                                                            | nodeB                                                                                            || expectedResult
+		GridTestData.nodeB                                                               | GridTestData.nodeB                                                                               || true
+		GridTestData.nodeB                                                               | GridTestData.nodeB.copy().build()                                                                || true
+		GridTestData.nodeB                                                               | GridTestData.nodeB.copy().vTarget(Quantities.getQuantity(1, PowerSystemUnits.PU)).build()        || true
+		GridTestData.nodeB                                                               | GridTestData.nodeB.copy().vTarget(Quantities.getQuantity(100, PowerSystemUnits.PERCENT)).build() || true
```

Comment:
Did you validate it against the old implementation? I did not, but as far as I understood the old implementation, diverging Units would lead to two objects not being equal. Please double check this.